src/hotspot/os/linux/os_linux.hpp

```diff
@@ -31,6 +31,7 @@
 
 class os::Linux {
   friend class CgroupSubsystem;
+  friend class CgroupUtil;
```

Comment:
why?

Reply:
Because `CgroupUtil::adjust_controller(CgroupCpuController* cpu)` uses `os::Linux::active_processor_count()` for the host cpu value which is protected.

Reply:
Okay. I dislike friend a bit since it undermines C++ scope protection. Why not make os::Linux::active_processor_count public then? If its needed from outside the Linux namespace, it should be public.

Side note, I assume we cannot just use `os::active_processor_count()` because it handles the containerized case, and what we really want here is the real physical count?

Not for this RFE, but a bit cleaner may be if we had something like os::active_processor_count() - taking care of handling the ActiveProcessorCount switch and reporting container limits if containerized - and something like "os::active_physical_processor_count()" , which would report the real physical number.

Reply:
I'd be happy to make `os::Linux::active_processor_count` public as part of this patch. Or add the ` "os::active_physical_processor_count()` api as you suggest. It would also make #19530 cleaner since it needs to know the host CPU values too. What would you like me to do here? How about keep as-is, then I'll file a new issue to clean this up by introducing `os::active_physical_processor_count()`?

Reply:
I leave it up to you, I am fine with doing it in this RFE or in a follow up.

Reply:
I've filed https://bugs.openjdk.org/browse/JDK-8339148 to do this. I'll do this clean-up with the introduction of the API.